libs/api/database/src/lib/content/stores/content.store.ts

```diff
@@ -278,36 +262,84 @@ export class ContentStore {
         // ...maybe work for another background queue processor
     }
 
+    //#endregion
+
+    //#region ---WORKS PUBLISHING---
+
     /**
-     * Finds content related to the the user's query.
-     * @param query The string the user searched for.
-     * @param pageNum The page of results to retrieve.
-     * @param maxPerPage The maximum number of results per page.
-     * @param filter The content filter to apply to returned results.
+     * Submits a piece of content to the approval queue.
+     *
+     * @param user The author of the content
+     * @param contentId The content to submit
      */
-    async findRelatedContent(
-        query: string,
-        kinds: ContentKind[],
-        pageNum: number,
-        maxPerPage: number,
-        filter: ContentFilter
-    ): Promise<PaginateResult<ContentDocument>> {
-        const paginateOptions: PaginateOptions = {
-            page: pageNum,
-            limit: maxPerPage,
-        };
-        const paginateQuery = {
-            $text: { $search: query },
-            'audit.published': PubStatus.Published,
+    async submitForApproval(user: JwtPayload, contentId: string): Promise<ContentDocument> {
+        const thisContent = await this.content.findOne({
+            _id: contentId,
+            author: user.sub,
             'audit.isDeleted': false,
-            kind: { $in: kinds },
-        };
-        this.determineContentFilter(paginateQuery, filter);
+        });
+
+        if (thisContent.kind !== ContentKind.PoetryContent && thisContent.stats.words < 750) {
+            throw new BadRequestException(`Content that isn't poetry needs to have a minimum word-count of 750. `);
+        }
+
+        await this.queue.addOneWork(contentId);
+        return await this.pendingWork(contentId, user.sub);
```

Comment:
This (and the below functions) extra bit of queue management is the kind of "not-quite strictly CRUD database specific" behavior that's riiight on the edge of making me say "this piece of logic belongs in a service". I think it's okay as it is, but it feels like something worth keeping an eye on.